proto/stream_plan.proto

```diff
@@ -707,6 +712,11 @@ enum DispatcherType {
   // piped into the downstream actor, if there are the same number of actors. If number of actors
   // are not the same, should use hash instead. Should be only used when distribution is the same.
   NO_SHUFFLE = 4;
+
+  // Dispatch by table name from upstream DB, used in CDC scenario which should has only one downstream actor.
+  // From the optimizer's point of view, it can be treated as a specialized version of HASH distribution
+  // that the hash key is the upstream table name.
+  CDC_TABLENAME = 5;
```

Comment:
Acturally I would prefer another approach to dispatching by the table name may require no changes on the dispatcher, that is, simply dispatch all data to the downstream with `NoShuffle`, then filter the data with a plain `Filter` before backfilling.

Reply:
The fragments after the data filtering is still possible to be hash-distributed. Only the `Filter` has to be singleton since there's actually no concrete data distribution at this stage.

The dispatch type should be `NoShuffle` because we don't want to exchange the full data (before filtering by table name) remotely.

Reply:
At first, I tried to use `Filter` to construct stream plan, but I found it is difficult to manually construct filter expression like `payload->>'source'->>'table'='table1'` in the optimizer. Then I was going to patch the `HASH`, and ended up adding a new dispatcher type according to comments. But it seems that introducing a new dispatcher type is intrusive to the framework. What about just patching the HASH to dispatch based on `downstream_table_name` as previously and refactor this part later? cc @BugenZhao  @fuyufjh

Reply:
I am okay with both approach.

By the way, even though @StrikeW did introduce a special kind of distribution, considering we may allow user to add more tables from the same upstream database, this might actually be simpler.

Reply:
To be honest, I don't see any need to hack the `Hash` dispatcher. 😕 It's just like a normal MV on MV, except that there's an extra `Filter` executor before the `Backfill`.

![image](https://github.com/risingwavelabs/risingwave/assets/25862682/8b1be114-e204-4ccf-a96d-c48e348e6caa)

> but I found it is difficult to manually construct filter expression like `payload->>'source'->>'table'='table1'` in the optimizer

I can understand, but it's at least self-contained and much easier to maintain compared to a new dispatcher type. So if possible, I would still suggest this approach.

Reply:
Nice diagram to demonstrate your point. I also want to make it more elegant and natural, let me beautify this part in next iteration (recorded https://github.com/risingwavelabs/risingwave/issues/13147). This PR is just a kickstart of the work, the work will no end at all.

Reply:
Oh, I got the problem. 

Previously I thought one dispatcher will distribute rows to all downstreams (tables), but I just realized that here one dispatcher corresponds to one downstream (table), right? This is indeed a bit weird, because the dispatcher is supposed to dispatch only, not dropping some rows.

So let me clarify my idea again: either 1) one dispatcher distribute all rows to multiple downstreams, or 2) bugen's idea: no_shuffle + filter , look good to me. But the current implementation needs some refactoring.

Reply:
> but I just realized that here one dispatcher corresponds to one downstream (table), right? This is indeed a bit weird, because the dispatcher is supposed to dispatch only, not dropping some rows.

Exactly. This is preferred long ago as https://github.com/risingwavelabs/risingwave/issues/2272, where one dispatcher is only responsible for a **single downstream fragment**, then a single downstream table. So that dropping MV on MV only drops a dispatcher, not mutates it.